### PR TITLE
Change links to index.jsp to point to application root #7405

### LINKED ISSUE #7405: Change links to index.jsp to point to application root
Example: the TEAMMATES logo links to the application root `/`, whereas the Home menu item links to `/index.jsp`. For consistency, all links to `/index.jsp` should be changed to point to `/`. Requires updating of test HTML files using GodMode.

Add: such links can be found at:
- `Const.java`
- `LogoutServlet.java`
- `bodyFooter.tag`
- `helpPage.tag`
- `staticPage.tag`

### PR DESCRIPTION
**Pull request by nikiforosbotis**
Fixes #7405 

**Outline of solution**

Found where is the tag which is used for every single jsp page that is linked with the application root (index.jsp) **(this page is the `staticPage.tag`)** and did the necessary change in the link.

Moreover, having read the document describing [GodMode](https://github.com/TEAMMATES/teammates/blob/master/docs/godmode.md), I turned the corresponding variable to "true" in the  `BaseUiTestCase.java` and then ran the tests having GodMode on.
godmode

### CODE REVIEW

**Comment by nikiforosbotis** [comment]
@wkurniawan07  Do you have any idea why these tests are failing? I am trying to understand but it seems like this link "\" is not understandable in a particular code snippet.

**Comment by whipermr5** [comment]
> Do you have any idea why these tests are failing?

@nikiforosbotis Do run all 24 tests indicated in the failing build log with GodMode enabled to update the HTML files.

> I am trying to understand but it seems like this link "\" is not understandable in a particular code snippet.

Can you elaborate on this?

**Comment by whipermr5** [comment]
@nikiforosbotis GodMode is meant to be used locally to update the test HTML files. After enabling it, run the 24 tests on your local machine and commit the changed HTML files. It is only when the updated HTML files are pushed that the build can pass.

**Comment by nikiforosbotis** [comment]
@whipermr5 I do run the tests via GodMode locally but nothing changes. What can I do?

**Comment by whipermr5** [comment]
@nikiforosbotis Try cleaning and building then restart the dev server and try again. Also check that you're not testing against a staging server.

**Comment by wkurniawan07** [comment]
@nikiforosbotis it looks like you're reverting a change that should be included in e314594?

**Comment by nikiforosbotis** [comment]
@wkurniawan07 I did this because I was wondering if this is the problem of the failing tests. I will revert the change.

**Comment by nikiforosbotis** [comment]
@whipermr5 I did re-run the tests using GodMode and after having cleaned and built but the results were the same. Nothing really changes and the most surprising thing is that the reasons behind each test failure seem to be different. The important thing is that the html files are not updated automatically. Do you have any idea on what to do now?

error

**Comment by whipermr5** [comment]
`npm run build`?

**Comment by nikiforosbotis** [comment]
@whipermr5 I did run the `npm run build`, some tests from the failed ones (12/34) did pass, but still the html files did not change.  After running the tests, I am doing `git commit -a -m` and nothing changes.

Can you help me specifying the problem?

Thank you!
a

**Comment by whipermr5** [comment]
@nikiforosbotis The GodMode switch is now in `test.properties`; did you switch it on there?

**Comment by nikiforosbotis** [comment]
@whipermr5 Will check it out again

**Comment by nikiforosbotis** [comment]
Dear @whipermr5,

Thank you very much for your help.
I did try again, doing the `npm run build ` and then turning GodMode to on from `test.properties` but nothing changed actually.

I have 25 failing tests that fail every time I run the test suite. The messaged from each test is like the one shown on the image. I guess that it means the html pages are not updated.

Is there a reason to fix that? Thank you.
a

**Comment by whipermr5** [comment]
@nikiforosbotis You did enable it in `test.properties` and not `test.template.properties` right?

You can also try running `GodModeTest`.

**Comment by nikiforosbotis** [comment]
@whipermr5 Thank you for your help. I still can't understand in which file you refer to in order to enable GodMode. I am searching for the corresponding field in the `test.properties` but I can't find it. I do find it only in the `test.template.properties`. Why that?

**Comment by whipermr5** [comment]
@nikiforosbotis Do read #7464 to understand :) In summary, you need to copy the option into `test.properties`.

**Comment by nikiforosbotis** [comment]
@whipermr5 As you can see, many tests ran with GodMode and several test files were automatically changed through this procedure. However, there are still some tests which fail continuously (I re-ran them several times). An indicative error that I get on these tests is this, as shown in the image. Do you have an idea about this? Thank you.

a

**Comment by whipermr5** [comment]
@nikiforosbotis There's something wrong with your Firefox setup; check the path in `test.properties`.

**Comment by whipermr5** [comment]
The failing tests show quite a few unrelated changes; do merge in the latest `master` and run `npm run build` before re-running the failed tests with GodMode enabled.

**Comment by nikiforosbotis** [comment]
@whipermr5 I will do it very soon and I will let you know

File: src/test/resources/pages/instructorStudentRecords.html [hunk 1]
```diff
@@ -193,7 +193,7 @@ <h4 class="modal-title">
     <div class="row">
       <div class="col-md-12">
         <br>
-        <div class="student_feedback panel panel-default" data-courseid="ISR.CS2104" data-fsname="First feedback session" data-googleid="ISR.teammates.test" data-studentemail="benny.c.tmms@gmail.tmt" id="studentFeedback-0">
+        <div class="student_feedback panel panel-default load-feedback-session" data-courseid="ISR.CS2104" data-fsname="First feedback session" data-googleid="ISR.teammates.test" data-studentemail="benny.c.tmms@gmail.tmt" id="studentFeedback-0">
```
Comment (whipermr5) [inline src/test/resources/pages/instructorStudentRecords.html]: This and almost all the other changes to this file are unrelated to the issue.

File: src/test/resources/pages/instructorStudentRecordsPageMixedQuestionType.html [hunk 2]
```diff
@@ -128,7 +128,7 @@ <h4 class="modal-title">
     <div class="row">
       <div class="col-md-12">
         <br>
-        <div class="student_feedback panel panel-default" data-courseid="FSQTT.idOfTypicalCourse1" data-fsname="CONSTSUM Session" data-googleid="FSQTT.idOfInstructor1OfCourse1" data-studentemail="student1InCourse1@gmail.tmt" id="studentFeedback-0">
+        <div class="student_feedback panel panel-default load-feedback-session" data-courseid="FSQTT.idOfTypicalCourse1" data-fsname="CONSTSUM Session" data-googleid="FSQTT.idOfInstructor1OfCourse1" data-studentemail="student1InCourse1@gmail.tmt" id="studentFeedback-0">
```
Comment (whipermr5) [inline src/test/resources/pages/instructorStudentRecordsPageMixedQuestionType.html]: This and almost all the other changes to this file are unrelated to the issue.

File: src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html [hunk 3]
```diff
@@ -128,7 +128,7 @@ <h4 class="modal-title">
     <div class="row">
       <div class="col-md-12">
         <br>
-        <div class="student_feedback panel panel-default" data-courseid="ISR.CS1101" data-fsname="Instructor feedback session" data-googleid="ISR.teammates.test" data-studentemail="teammates.test@gmail.tmt" id="studentFeedback-0">
+        <div class="student_feedback panel panel-default load-feedback-session" data-courseid="ISR.CS1101" data-fsname="Instructor feedback session" data-googleid="ISR.teammates.test" data-studentemail="teammates.test@gmail.tmt" id="studentFeedback-0">
```
Comment (whipermr5) [inline src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html]: This and almost all the other changes to this file are unrelated to the issue.

Comment (nikiforosbotis) [inline src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html]: @whipermr5 How can I revert them? manually?

Comment (whipermr5) [inline src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html]: I'd suggest reverting the entire file first, then re-running `InstructorStudentRecordsPageUiTest` with GodMode enabled. The only lines that should change are those containing links to `index.jsp`. Any other unrelated change indicates a problem with your config somewhere - perhaps the issue lies elsewhere, but do merge the latest `master` and do `npm run build` again before re-running the test just in case.

Comment (nikiforosbotis) [inline src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html]: @whipermr5 Also, do you have any idea why these unrelated changes took place?

Comment (whipermr5) [inline src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html]: Might be due to recent changes to the JavaScript here: https://github.com/TEAMMATES/teammates/commit/cf7055ddad9c4414cd844844567fe39e0961783a#diff-0a81914fff38a10025f44b835e853fe2

Did you disable the cache in Firefox?

Comment (nikiforosbotis) [inline src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html]: Yes, I have disabled it. I am doing one more attempt by merging the latest master and the other step you propose.

Comment (nikiforosbotis) [inline src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html]: @whipermr5 The unrelated changes should be reverted manually? Because if i revert the whole commit, all changes will be reverted (included those which are needed).

Comment (whipermr5) [inline src/test/resources/pages/instructorStudentRecordsPageWithPrivateFeedback.html]: Yes 👍

**Comment by whipermr5** [comment]
@nikiforosbotis Nearly there! Only one more test failing - `AdminHomePageUiTest `. After the GodMode run, do disable GodMode and run the test locally again to ensure it passes (this is actually a necessary step as the GodMode run can result in false positives, as documented [here](https://github.com/TEAMMATES/teammates/blob/master/docs/godmode.md#final-notes)). Only commit and push once it passes locally to save time waiting for Travis to run the entire test suite when we're only concerned with `AdminHomePageUiTest`.

**Comment by nikiforosbotis** [comment]
@whipermr5 OK, got it. I will do it asap and will let you know about the results. Thank you for your support.

**Comment by nikiforosbotis** [comment]
@whipermr5 I did try to run the last failing test `AdminHomePageUiTest` but after several attempts, it still fails, as shown in the picture below.
Do you have any idea why is this happening?

a

**Comment by whipermr5** [comment]
Is your dev server running? Look at the dev server logs for errors.

**Comment by nikiforosbotis** [comment]
@whipermr5 Do you mean this?

a

**Comment by whipermr5** [comment]
No; I meant the other console showing the dev server running.

**Comment by nikiforosbotis** [comment]
@whipermr5 Here you go:

a
b

**Comment by whipermr5** [comment]
I think you need to `npm run build`

**Comment by nikiforosbotis** [comment]
I did it but I have some problems even though until yesterday (before I merge the current master) it was being built without any problems. Do you want me to send you the full log?

a

**Comment by whipermr5** [comment]
`npm install` first

**Comment by nikiforosbotis** [comment]
@whipermr5 I had already installed it since I was doing `npm run build` every time I was asked to do so. But probably something changed. After doing them all again, I'm getting this message and the test still fails.
a

**Comment by whipermr5** [comment]
The test error message should be different now though; do post it here.

**Comment by nikiforosbotis** [comment]
@whipermr5 The test still fails and the message is `TimeoutException: Timed out after 15 seconds waiting for element to be clickable (...)`

The dev server's log is this:
a
b
c
d
e
f

**Comment by nikiforosbotis** [comment]
And the test failure exception:
a

**Comment by whipermr5** [comment]
@nikiforosbotis That looks like a temporary error; re-run the test a few times and if it still fails, increase the value of `test.timeout` in your `test.properties`.

**Comment by whipermr5** [comment]
@nikiforosbotis I pulled your branch and the test passed. I think it's ready for Travis! 👍

**Comment by nikiforosbotis** [comment]
@whipermr5 It seems that it is now fine. Do you agree?

**Comment by whipermr5** [comment]
@nikiforosbotis I found one more file to be updated: `sitemap.xml`

**Comment by nikiforosbotis** [comment]
@whipermr5 Is it fine now?

**Comment by nikiforosbotis** [comment]
@whipermr5 Thank you! Thank you also for the support and the very detailed instructions.
